lib/ParseSIL/ParseSIL.cpp

```diff
@@ -7094,14 +7111,10 @@ bool SILParserTUState::parseSILDifferentiabilityWitness(Parser &P) {
       return true;
   }
 
-  auto origFnType = originalFn->getLoweredFunctionType();
-  auto *parameterIndexSet = IndexSubset::get(
-      P.Context, origFnType->getNumParameters(), parameterIndices);
-  auto *resultIndexSet = IndexSubset::get(
-      P.Context, origFnType->getNumResults(), resultIndices);
-  SILDifferentiabilityWitness::create(
-      M, *linkage, originalFn, parameterIndexSet, resultIndexSet,
-      derivativeGenSig, jvp, vjp, isSerialized);
+  SILDifferentiabilityWitness::createDefinition(
+      M, linkage ? *linkage : SILLinkage::DefaultForDefinition, originalFn,
```

Comment:
Logic above always sets `linkage`:
```cpp
  // Parse the linkage.
  Optional<SILLinkage> linkage;
  if (parseSILLinkage(linkage, P))
    return true;
  // Default to public linkage.
  if (!linkage)
    linkage = SILLinkage::Public;
```

How about passing `*linkage` to `SILDifferentiabilityWitness::createDefinition`, but using `SILLinkage::DefaultForDefinition` instead of `SILLinkage::Public` above? (`DefaultForDefinition` is equal to `Public`)

```
  if (!linkage)
    linkage = SILLinkage:: DefaultForDefinition;
```

Reply:
My bad for overlooking, thanks!